File: Seminar_06/Model/ComplexModel/MultComplex.java
package Seminar_06.Model.ComplexModel;

import Seminar_06.Model.CalcModel;

/**
 * Класс для рассчёта произведения комплексных чисел
 */
public class MultComplex extends CalcModel {
    @Override
    public double result(double x, double y) {
        return x * y;
    }

    /**
     * Метод рассчёта результата
     */
    public Complex resultComplex(double x1, double x2, double y1, double y2) {
        double newReal = result(x1, x2) - result(y1, y2);
        double newImage = result(y1, x2) + result(x1, y2);
        Complex result = new Complex(newReal, newImage);
        return result;
    }
}

File: Seminar_07/Model/Complex_model/MultComplex.py
import Seminar_07.Model.Rational_model.MultRational as Mult
import Seminar_07.Model.Complex_model.ComplexNumber as ComplexNumber


class MultComplex():
    '''Умножение'''

    def result(self, x1, x2, y1, y2):
        new_real = Mult.MultRational.result(
            self, x1, x2) - Mult.MultRational.result(self, y1, y2)
        new_image = Mult.MultRational.result(
            self, y1, x2) + Mult.MultRational.result(self, x1, y2)
        result = ComplexNumber.Complex(new_real, new_image)
        return result
